app/code/Magento/CatalogImportExport/Model/Import/Product.php

```diff
@@ -1737,7 +1749,43 @@ protected function _saveProducts()
                 $position = 0;
                 foreach ($rowImages as $column => $columnImages) {
                     foreach ($columnImages as $columnImageKey => $columnImage) {
-                        if (!isset($uploadedImages[$columnImage])) {
+                        $hash = '';
+                        if (filter_var($columnImage, FILTER_VALIDATE_URL) === false) {
+                            $filename = $importDir . DIRECTORY_SEPARATOR . $columnImage;
+                            if ($this->fileDriver->isExists($filename)) {
+                                $hash = hash_file(
```

Comment:
I would recommend wrapping the hash_file in a private function and create a private constant with the 'sha256' algorithm, we are using this 3 times on this PR.